app/vmalert/config/s3/s3.go

```diff
@@ -0,0 +1,150 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+// FS represents filesystem for backups in S3.
+//
+// Init must be called before calling other FS methods.
+type FS struct {
+	// Path to S3 credentials file.
+	CredsFilePath string
+
+	// Path to S3 configs file.
+	ConfigFilePath string
+
+	// GCS bucket to use.
+	Bucket string
+
+	// Prefix is the prefix filter to query objects
+	// whose names begin with this prefix.
+	//
+	// Optional.
+	Prefix string
+
+	// Set for using S3-compatible enpoint such as MinIO etc.
+	CustomEndpoint string
+
+	// Force to use path style for s3, true by default.
+	S3ForcePathStyle bool
+
+	// The name of S3 config profile to use.
+	ProfileName string
+
+	s3 *s3.Client
+}
+
+// Init initializes fs.
+//
+// The returned fs must be stopped when no long needed with MustStop call.
+func (fs *FS) Init() error {
+	if fs.s3 != nil {
+		logger.Panicf("BUG: Init is already called")
+	}
+	configOpts := []func(*config.LoadOptions) error{
+		config.WithSharedConfigProfile(fs.ProfileName),
+		config.WithDefaultRegion("us-east-1"),
```

Comment:
Should this be as const, or can we make it configurable?

Reply:
Default region is needed for buckets auto-discovery. It main be a problem at Chinese region, but I'm not sure about that.